File: src/main/java/com/github/sshaddicts/lucrecium/util/FileInteractions.java
package com.github.sshaddicts.lucrecium.util;

import org.apache.commons.io.FileUtils;
import org.opencv.core.Mat;
import org.opencv.imgcodecs.Imgcodecs;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;


public class FileInteractions {

    private static Logger log = LoggerFactory.getLogger(FileInteractions.class);

    public static void saveMats(List<Mat> list, String directory) {
        for (Mat mat : list) {
            saveMat(mat, directory);
        }
    }

    public static void saveMat(Mat mat, String directory) {
        File f = new File(directory);

        if(!f.exists()){
            if(!f.mkdir()){
                throw new IllegalArgumentException("directory is hacked: " + directory);
            }
        }
        Imgcodecs.imwrite(directory + "/image_" + System.currentTimeMillis() + ".png", mat);
    }

    public static List<String> getFileNamesFromDir(String parentDir) {
        List<String> filenames = new ArrayList<>();

        File parentDirectory = FileUtils.getFile(parentDir);
        File[] files = parentDirectory.listFiles();

        assert files != null;
        for (File file : files) {
            if (file.isFile()) {
                filenames.add(parentDir + "/" + file.getName());
            }
        }
        return filenames;
    }

    public static void saveMatWithName(Mat mat, String directory, String filename) {
        Imgcodecs.imwrite(directory + "/" + directory + "_" + filename + ".png", mat);
    }
}

File: src/main/java/com/github/sshaddicts/lucrecium/util/FileInteractions.kt
package com.github.sshaddicts.lucrecium.util

import org.apache.commons.io.FileUtils
import org.opencv.core.Mat
import org.opencv.imgcodecs.Imgcodecs
import org.slf4j.LoggerFactory
import java.io.File
import java.util.*


object FileInteractions {

    private val log = LoggerFactory.getLogger(FileInteractions::class.java)

    fun saveMats(list: List<Mat>, directory: String) {
        for (mat in list) {
            saveMat(mat, directory)
        }
    }

    fun saveMat(mat: Mat, directory: String) {
        val f = File(directory)

        if (!f.exists() && !f.mkdir()) {
            throw IllegalArgumentException("directory is hacked: " + directory)
        }

        Imgcodecs.imwrite(directory + "/image_" + System.nanoTime() + ".png", mat)
    }

    fun getFileNamesFromDir(parentDir: String): List<String> {
        val filenames = ArrayList<String>()

        val parentDirectory = FileUtils.getFile(parentDir)
        val files = parentDirectory.listFiles()!!

        for (file in files) {
            if (file.isFile) {
                filenames.add(parentDir + "/" + file.name)
            }
        }
        return filenames
    }

    fun saveMatWithName(mat: Mat, directory: String, filename: String) {
        Imgcodecs.imwrite(directory + "/" + directory + "_" + filename + ".png", mat)
    }
}
